source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -169,6 +170,12 @@ void AuthenticatorImpl::startVerify() {
   // isIssuerSpecified() check already make sure the issuer is in the cache.
   ASSERT(jwks_data_ != nullptr);
 
+  auto cache_jwt = jwks_data_->getTokenCache()->find(curr_token_->token(), jwt_cache_hited_);
+  if (jwt_cache_hited_) {
+    ENVOY_LOG(debug, "{}: verified token with cache hit: tokens size {}", name(), tokens_.size());
+    jwt_.reset(cache_jwt);
```

Comment:
Not good.  If you do this, cache_jwt will be owned by jwt_.   It will free cache_jwt when it is gone.   Do you want that?

Reply:
I don't see changes to skip jwtParsing and verification for cache hit.

Reply:
PTAL